e2e/runner/report.go

```diff
@@ -53,3 +60,70 @@ func (runner *E2ERunner) PrintTestReports(tr TestReports) {
 	}
 	runner.Logger.PrintNoPrefix(table)
 }
+
+// NetworkReport is a struct that contains the report for the network used after running e2e tests
+// This report has been initialized to check the emissions pool balance and if the pool is decreasing
+// TODO: add more complete data and validation to the network
+// https://github.com/zeta-chain/node/issues/1873
+type NetworkReport struct {
+	EmissionsPoolBalance sdkmath.Int
+	Height               uint64
+	CctxCount            int
+}
+
+// Validate validates the network report
+// This method is used to validate the network after running e2e tests
+// It checks the emissions pool balance and if the pool is decreasing
+func (nr NetworkReport) Validate() error {
+	if nr.EmissionsPoolBalance.GTE(sdkmath.NewIntFromBigInt(EmissionsPoolFunding)) {
+		return fmt.Errorf(
+			"emissions pool balance is not decreasing, expected less than %s, got %s",
+			EmissionsPoolFunding,
+			nr.EmissionsPoolBalance,
+		)
+	}
+	return nil
+}
+
+// GenerateNetworkReport generates a report for the network used after running e2e tests
+func (runner *E2ERunner) GenerateNetworkReport() (NetworkReport, error) {
+	// get the emissions pool balance
+	balanceRes, err := runner.BankClient.Balance(runner.Ctx, &banktypes.QueryBalanceRequest{
+		Address: txserver.EmissionsPoolAddress,
+		Denom:   config.BaseDenom,
+	})
+	if err != nil {
+		return NetworkReport{}, err
+	}
+	emissionsPoolBalance := balanceRes.Balance
+
+	// fetch the height and number of cctxs, this gives a better idea on the activity of the network
+
+	// get the block height
+	blockRes, err := runner.ZevmClient.BlockNumber(runner.Ctx)
+	if err != nil {
+		return NetworkReport{}, err
+	}
+
+	// get the number of cctxs
+	cctxsRes, err := runner.CctxClient.CctxAll(runner.Ctx, &crosschaintypes.QueryAllCctxRequest{})
```

Comment:
We should also add  len(tssHistory) as to this struct as that would have ballots associated with it as well . 

Can be added to the issue as description as well , because we need it only for the validation logic 
/ https://github.com/zeta-chain/node/issues/1873

Reply:
Updated the issue with tssHistory

>  as to this struct as that would have ballots associated with it as well .

What do you mean by this?

Reply:
Typo 
Meant that each tss would atleast 1 ballot associated with it , which would affect rewards distribution . 
Update to the issue seems fine for now